tests/user_metadata_test.go

```diff
@@ -0,0 +1,130 @@
+	id := "functional-user-metadata-SignalWithStartWorkflowExecution"
+	metadata := prepareTestUserMetadata()
+	request := &workflowservice.SignalWithStartWorkflowExecutionRequest{
+		RequestId:    uuid.New(),
+		Namespace:    s.namespace,
+		WorkflowId:   id,
+		WorkflowType: &commonpb.WorkflowType{Name: "functional-user-metadata-workflow-type"},
+		TaskQueue:    &taskqueuepb.TaskQueue{Name: "functional-user-metadata-taskqueue", Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+		SignalName:   "TEST-SIGNAL",
+		UserMetadata: metadata,
+	}
+
+	we, err := s.engine.SignalWithStartWorkflowExecution(NewContext(), request)
+	s.NoError(err)
+
+	// Verify that the UserMetadata associated with the start event is returned in the describe response.
+	describeInfo, err := getDescribeWorkflowExecutionInfo(s.engine, s.namespace, id, we.RunId)
+	s.NoError(err)
+	s.NotNil(describeInfo.ExecutionConfig.UserMetadata)
+	s.EqualExportedValues(describeInfo.ExecutionConfig.UserMetadata, metadata)
+}
+
+// TestUserMetadataExecuteMultiOperation issues a multi operation that includes start workflow with metadata attached.
+// It verifies that the sent metadata is correctly recorded and returned in the describe workflow response.
+func (s *FunctionalSuite) TestUserMetadataExecuteMultiOperation() {
+	id := "functional-user-metadata-ExecuteMultiOperation"
+	metadata := prepareTestUserMetadata()
+	startWorkflowRequest := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:    uuid.New(),
+		Namespace:    s.namespace,
+		WorkflowId:   id,
+		WorkflowType: &commonpb.WorkflowType{Name: "functional-user-metadata-workflow-type"},
+		TaskQueue:    &taskqueuepb.TaskQueue{Name: "functional-user-metadata-taskqueue", Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+		UserMetadata: metadata,
+	}
+	updateWorkflowRequest := &workflowservice.UpdateWorkflowExecutionRequest{
+		Namespace:         s.namespace,
+		WorkflowExecution: &commonpb.WorkflowExecution{WorkflowId: id},
+		Request: &updatepb.Request{
+			Meta:  &updatepb.Meta{UpdateId: "UPDATE_ID"},
+			Input: &updatepb.Input{Name: "NAME"},
+		},
+	}
+	request := &workflowservice.ExecuteMultiOperationRequest{
+		Namespace: s.namespace,
+		Operations: []*workflowservice.ExecuteMultiOperationRequest_Operation{
+			{ // start workflow operation
+				Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+					StartWorkflow: startWorkflowRequest,
+				},
+			},
+			{ // update workflow operation
+				Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_UpdateWorkflow{
+					UpdateWorkflow: updateWorkflowRequest,
+				},
+			},
+		},
+	}
+
+	_, err := s.engine.ExecuteMultiOperation(NewContext(), request)
+	s.NoError(err)
+
+	// Verify that the UserMetadata associated with the start event is returned in the describe response.
+	describeInfo, err := getDescribeWorkflowExecutionInfo(s.engine, s.namespace, id, "")
+	s.NoError(err)
+	s.NotNil(describeInfo.ExecutionConfig.UserMetadata)
```

Comment:
What happens if `UserMetadata` is nil, wouldn't `EqualExportedValues` catch that?